aztec_gddt/params.py

```diff
@@ -80,14 +85,130 @@
+single_shock_gas_fee_l1_time_series[-final_time:] = steady_gas_fee_l1_time_series[-final_time:].copy()
+single_shock_gas_fee_l1_time_series[initial_time:TIMESTEPS - final_time] = steady_gas_fee_l1_time_series[initial_time:TIMESTEPS - final_time].copy() + L1_SHOCK_AMOUNT
+
+single_shock_gas_fee_blob_time_series[0:initial_time] = steady_gas_fee_blob_time_series[0:initial_time].copy()
+single_shock_gas_fee_blob_time_series[-final_time:] = steady_gas_fee_blob_time_series[-final_time:].copy()
+single_shock_gas_fee_blob_time_series[initial_time:TIMESTEPS - final_time] = steady_gas_fee_blob_time_series[initial_time:TIMESTEPS - final_time].copy() + L1_SHOCK_AMOUNT
+
+
+def single_shock_l1_gas_estimate(state: AztecModelState):
+    if state['timestep'] < len(single_shock_gas_fee_l1_time_series):
+        return single_shock_gas_fee_l1_time_series[state['timestep']]
+    else:
+        return single_shock_gas_fee_l1_time_series[-1]
+
+def single_shock_blob_gas_estimate(state: AztecModelState):
+    if state['timestep'] < len(single_shock_gas_fee_blob_time_series):
+        return single_shock_gas_fee_blob_time_series[state['timestep']]
+    else:
+        return single_shock_gas_fee_blob_time_series[-1]
+
+SINGLE_SHOCK_GAS_ESTIMATORS = L1GasEstimators(
+    proposal= single_shock_l1_gas_estimate,
+    commitment_bond= single_shock_l1_gas_estimate,
+    content_reveal= single_shock_l1_gas_estimate,
+    content_reveal_blob= single_shock_blob_gas_estimate, # NOTE: this is a HACK assumption, gas was estimated from various documents by Aztec Labs
+    rollup_proof= single_shock_l1_gas_estimate
+)
+
+#############################################################
+## End: single shock gas estimators defined                ##
+#############################################################
+
+#############################################################
+## Begin: intermittent shock gas estimators defined        ##
+#############################################################
+
+L1_INTER_SHOCK_AMPLITUDE = 100 # Amplitude of wave
+L1_INTER_SHOCK_PERIOD = 10 # Period of wave
+num_points = (TIMESTEPS - final_time) - initial_time
+t = np.arange(initial_time, initial_time + num_points)
+
+raw_shock_signal = L1_INTER_SHOCK_AMPLITUDE * np.sin(2 * np.pi * t / L1_INTER_SHOCK_PERIOD) + L1_INTER_SHOCK_AMPLITUDE
+L1_INTER_SHOCK_SIGNAL = np.array([floor(max(x,0)) for x in raw_shock_signal])
+
+
+intermit_shock_gas_fee_l1_time_series = np.zeros(TIMESTEPS)
+intermit_shock_gas_fee_blob_time_series = np.zeros(TIMESTEPS)
+
+intermit_shock_gas_fee_l1_time_series[0:initial_time] = steady_gas_fee_l1_time_series[0:initial_time].copy()
+intermit_shock_gas_fee_l1_time_series[-final_time:] = steady_gas_fee_l1_time_series[-final_time:].copy()
+intermit_shock_gas_fee_l1_time_series[initial_time:TIMESTEPS - final_time] = steady_gas_fee_l1_time_series[initial_time:TIMESTEPS - final_time].copy() + L1_INTER_SHOCK_SIGNAL
+
+def intermit_shock_l1_gas_estimate(state: AztecModelState):
```

Comment:
note: the above estimators are relevant for the Gas Fee, not for the Gas Units per Tx